### ISSUE TITLE
How to use custom keychain group in brokered authentication

### ISSUE THREAD

**Issue by tobihagemann** [AzureAD/microsoft-authentication-library-for-objc#726]
This is somehow related to #281 and #282.

I noticed that you cannot set a custom keychain group when using brokered authentication. If you look at [`MSIDBrokerInteractiveController.m#L74`](https://github.com/AzureAD/microsoft-authentication-library-common-for-objc/blob/3a5b7250ca293e4aa44abdf12284ad9faa93e2f0/IdentityCore/src/controllers/ios/MSIDBrokerInteractiveController.m#L74), it seems like that you can set `keychainAccessGroup` by passing it via `parameters` of type `MSIDInteractiveRequestParameters` (or actually in the superclass [`MSIDRequestParameters.h#L76`](https://github.com/AzureAD/microsoft-authentication-library-common-for-objc/blob/3a5b7250ca293e4aa44abdf12284ad9faa93e2f0/IdentityCore/src/parameters/MSIDRequestParameters.h#L76)).

However, if you look closely to how this is being called, you'll see that in [`MSALPublicClientApplication.m#L854-L1003`](https://github.com/AzureAD/microsoft-authentication-library-for-objc/blob/94b17154f58a732691bfdcd768fbf2563b09c41f/MSAL/src/MSALPublicClientApplication.m#L854-L1003) the `msidParams.keychainAccessGroup` is never being set.

## Solution 1
Add the following line to somewhere in [`MSALPublicClientApplication.m#L784-L1005`](https://github.com/AzureAD/microsoft-authentication-library-for-objc/blob/94b17154f58a732691bfdcd768fbf2563b09c41f/MSAL/src/MSALPublicClientApplication.m#L784-L1005):

```objc
msidParams.keychainAccessGroup = self.internalConfig.cacheConfig.keychainSharingGroup;
```

## Solution 2
Add a new `keychainAccessGroup` property to  [`MSALInteractiveTokenParameters`](https://github.com/AzureAD/microsoft-authentication-library-for-objc/blob/dev/MSAL/src/public/MSALInteractiveTokenParameters.h) and add the following line similar to solution 1:

```objc
msidParams.keychainAccessGroup = parameters.keychainAccessGroup;
```

## Solution 3
Add [`MSIDKeychainTokenCache.h`](https://github.com/AzureAD/microsoft-authentication-library-common-for-objc/blob/3a5b7250ca293e4aa44abdf12284ad9faa93e2f0/IdentityCore/src/cache/MSIDKeychainTokenCache.h) to the public header files in the Podspec so that the `defaultKeychainGroup` can be set.

---

I'd be happy to make a PR. :rocket:

Edit: In my opinion, solution 1 makes the most sense because you'd have only "one" location (in the cache config) to set a custom keychain group.

**Comment by oldalton** [comment]
Thanks, fixed in https://github.com/AzureAD/microsoft-authentication-library-for-objc/pull/728.

**Comment by tobihagemann** [comment]
Awesome, thank you!

**Comment by oldalton** [comment]
Should be fixed in 1.0.1 that is now released. Please reopen the issue if it still persists in 1.0.1.
